app/controllers/account_reset/cancel_controller.rb

```diff
@@ -1,9 +1,24 @@
 module AccountReset
   class CancelController < ApplicationController
+    before_action :check_feature_enabled
+
+    def show
+      return render :show unless token
```

Comment:
If there isn't a token, we want to know about it, and display a helpful message to the user. This keeps it consistent with what `AccountReset::Cancel` does. Since the token validation now happens in two classes, it probably makes sense to extract it into a validator as Jim suggested, and include it in both classes.

Reply:
Agreed.  I'll need to refactor cancel now to use the validator

Reply:
I just copied what DeleteAccountController was doing and it renders a missing token with no message.  They will get an error message either way because the post won't work.

Reply:
Yeah, I realized after I made that comment that this is because of the whole remove the token from the URL thing.